lib/puppet/provider/docker_compose/ruby.rb

```diff
@@ -17,11 +17,21 @@
     environment(HOME: '/root')
   end
 
+  def set_tmpdir
+    return unless resource[:tmpdir]
+    # Check if the the tmpdir target exists
+    Puppet.warning("#{resource[:tmpdir]} (defined as docker_compose tmpdir) does not exist") unless Dir.exist?(resource[:tmpdir])
+    # Set TMPDIR environment variable only if defined among resources and exists
+    ENV['TMPDIR'] = resource[:tmpdir] if Dir.exist?(resource[:tmpdir])
```

Comment:
How/when is this variable reset to it's original value?

I suspect that this would have an impact with other types / provider depending on resource ordering:

```puppet
foo { 'before': # ENV['TMPDIR'] #=> nil
}

docker_compose { 'bar':
  path => '/here/or/there',
}

foo { 'after': # ENV['TMPDIR'] #=>  '/here/or/there'
}
```

Reply:
@smortex the variable only appears to have effect within the resource, and also the resource collector in init.pp ensures that docker_compose runs lasts, so my tests to put (echo) the ENV['TMPDIR'] during run show it as unset or reflect the system variable within prior resources
```
    Class['docker::repos']
    -> Class['docker::install']
    -> Class['docker::config']
    -> Class['docker::service']
    -> Docker::Registry <||>
    -> Docker::Image <||>
    -> Docker::Run <||>
    -> Docker_compose <||>
```

Reply:
@canihavethisone Thanks for that explanation. Makes sense to me.

@smortex What do you think?